packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -1,11 +1,19 @@
 import path from 'path';
 import type * as ts from 'typescript';
 
-interface FileSpecifier {
+type FileSpecifier = {
   from: 'file';
   name: string | string[];
-  path?: string;
-}
+} & (
+  | {
+      path?: string;
+      excludePaths?: undefined;
+    }
+  | {
+      path?: undefined;
+      excludePaths?: string | string[]; // defaults to `["node_modules"]`.
+    }
+);
```

Comment:
Starting a new conversation here: why make `excludePaths` and `path` mutually exclusive? What if someone wants to provide both? E.g.:

```js
{
  path: "examples",
  excludePaths: ["examples/*/node_modules"]
}
```

Proposal: can we switch this back to a straightforward interface?

```ts
interface FileSpecifier {
  from: 'file';
  name: string | string[];
  path?: string;
  excludePaths?: string | string[]; // defaults to `["node_modules"]`.
}
```